python/cuml/cuml/metrics/_classification.py

```diff
@@ -13,16 +13,121 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
+import warnings
 
-from cuml.internals.input_utils import input_to_cupy_array
 import cuml.internals
+from cuml.internals.input_utils import input_to_cupy_array
 from cuml.internals.safe_imports import cpu_only_import
 from cuml.internals.safe_imports import gpu_only_import
 
 cp = gpu_only_import("cupy")
+cudf = gpu_only_import("cudf")
 np = cpu_only_import("numpy")
 
 
+def _input_to_cupy_or_cudf_series(x, check_rows=None):
+    """Coerce the input to a 1D cupy array or cudf Series.
+
+    For classification problems we need to support the full range
+    of supported input dtypes. cupy cannot support string labels,
+    and cudf cannot support float16. To handle this, we prefer cudf
+    if the input is cudf, otherwise try to coerce to cupy, falling
+    back to cudf if the dtype isn't supported.
+    """
+    if isinstance(x, cudf.Series):
+        # Drop the index so comparisons don't try to align on index
+        out = x.reset_index(drop=True)
+        n_cols = 1
+    else:
+        try:
+            out, _, n_cols, _ = input_to_cupy_array(x)
+            out = out.squeeze()  # ensure 1D
+        except ValueError:
+            # Unsupported dtype, use cudf instead
+            # Drop the index so comparisons don't try to align on index
+            out = cudf.Series(x, nan_as_null=False, copy=False).reset_index(
+                drop=True
+            )
+            n_cols = 1
+
+    n_rows = len(out)
+
+    if n_cols > 1:
+        raise ValueError(f"Expected 1 columns but got {n_cols} columns.")
+    if check_rows is not None and n_rows != check_rows:
+        raise ValueError(f"Expected {check_rows} rows but got {n_rows} rows.")
+
+    return out
+
+
+@cuml.internals.api_return_any()
+def accuracy_score(
+    y_true, y_pred, *, sample_weight=None, normalize=True, **kwargs
```

Comment:
We now have _almost_ full compatibility with `sklearn` here. We don't support the multi-output cases here since those require a bunch more infra around detecting label kinds (we don't support those in cuml anyway, so not a big deal). In the long run I hope we can just rely on the implementations in sklearn itself once their array-api support solidifies.